### ISSUE TITLE
Examine de.genes after SingleR using multiple references

### ISSUE THREAD

**Issue by akramdi** [LTLA/SingleR#121]
Hi, 

I have just ran singleR using multiple references at once. I wanted to examine the expression of marker genes using `plotHeatmap()` but I can't correctly extract the de.genes for each label. Here are the commands I used:

```
#---- Run singleR
method="single"
level="main"
refData=list(ref1,ref2,ref4,ref5)
refLabels=list(ref1$label.main,ref2$label.main,ref4$label.main,ref5$label.main)
preds<- SingleR(test=testData, ref=refData, labels=refLabels, method=method)

#---- Extract de.genes
> unique(preds$labels)
 [1] "Adipocytes"           "Neurons"              "Macrophages"         
 [4] "Fibroblasts"          "Endothelial cells"    "HSC"                 
 [7] "NK cells"             "Mesangial cells"      "Neuroepithelial_cell"
[10] "Erythrocytes"         "CD8+ T-cells"         "Monocytes"           
[13] "Dendritic cells"      "Myocytes"             "CD4+ T-cells"        
[16] "Astrocyte"            "Epithelial cells"     "Smooth muscle"       
[19] "Skeletal muscle"      "B-cells"              "iPS_cells"           
[22] "B_cell"               "Gametocytes"          "Endothelial_cells"   
[25] "Melanocytes"          "Chondrocytes"         "DC"                  
[28] "B cells"              "Neutrophils"          "Myelocyte"           
[31] "Erythroblast"         "Macrophage"           "Pre-B_cell_CD34-"    
[34] "Pro-Myelocyte"        "NK_cell"              "Epithelial_cells"    
[37] "Erythroid cells"      "Keratinocytes"        "Embryonic_stem_cells"
[40] "Basophils"            "CMP"                  "Pro-B_cell_CD34+"    
[43] "MEP"                 

all.markers <- metadata(preds)$de.genes

#--- get de.genes for label Neurons
> all.markers[["Neurons"]]
NULL
```

However, this works:
```
all.markers[[1]][["Neurons"]]
all.markers[[2]][["Neurons"]]
all.markers[[3]][["Neurons"]]
....
```

When I run singleR using only one reference, I can extract de.genes using `all.markers[[label]]`. How can I do the same here? I don't understand why is there more than location for a label.

Ultimately, I'd like to loop over labels and plot the expression heatmap for markers to check if the annotations make sens.

Many thanks for the help!

**Comment by LTLA** [comment]
**tl;dr** The documentation is wrong because I forgot to update it. I'll answer this question in #122.
